4651 : adding a change to fix database is locked error

RADXIshan opened the pull request:
Made changes in the file cvedb.py to fix the issue 4651

stvml commented on cve_bin_tool/cvedb.py:
```diff
@@ -763,18 +765,23 @@ def update_vendors(self, cve_data):
-            LOGGER.error("Database cursor does not exist")
-            raise CVEDBError
-        return cursor
+    def db_open_and_get_cursor(self):
```
We should keep the type hint here.

stvml commented on cve_bin_tool/cvedb.py:
```diff
@@ -763,18 +765,23 @@ def update_vendors(self, cve_data):
+    retries = 0
+    while retries < MAX_RETRIES:
+        try:
+            self.connection = sqlite3.connect(self.dbpath, timeout=10)  # Extend timeout
```
I'm not confident we should extend the timeout in addition to adding the retry logic, but I could be convinced. What's the thinking behind this change?

stvml commented on cve_bin_tool/cvedb.py:
```diff
@@ -763,18 +765,23 @@ def update_vendors(self, cve_data):
+            if "database is locked" in str(e):
+                retries += 1
+                wait_time = RETRY_DELAY * (2 ** retries)  # Exponential backoff
+                print(f"Database locked, retrying in {wait_time:.2f} seconds...")
```
Please make use of the `LOGGER` object here instead of a bare `print` call.

stvml commented on cve_bin_tool/cvedb.py:
```diff
@@ -763,18 +765,23 @@ def update_vendors(self, cve_data):
+                print(f"Database locked, retrying in {wait_time:.2f} seconds...")
+                time.sleep(wait_time)
+            else:
+                raise  # Raise other errors
```
I think for this `raise` and the one right below it, we should use `LOGGER.error` and `raise CVEDBError` instead of raising whatever `sqlite3` gives us. That will keep the existing error handler logic usable if it's ever needed for something.

stvml commented on cve_bin_tool/cvedb.py:
```diff
@@ -763,18 +765,23 @@ def update_vendors(self, cve_data):
+    while retries < MAX_RETRIES:
+        try:
+            self.connection = sqlite3.connect(self.dbpath, timeout=10)  # Extend timeout
+            self.connection.execute("PRAGMA journal_mode=WAL;")  # Improve concurrency
```
Trying to think of side effects this might cause. This will create some extra files on disk next to our database, right? I don't think there would be issues with this change, but I defer to the other maintainers who might be more familiar with the SQLite stuff...

stvml commented:
Thanks for working on this, @RADXIshan! I think this could help with the intermittent database issues. Please look over my review comments when you get a chance.

RADXIshan commented:
Thanks for the review, I will look into the issues you have mentioned and
try to fix them . I would be happy to collaborate with you on this project
and i really appreciate some guidance as I have just started getting into
open source.

On Fri, 28 Mar 2025 at 11:46 PM, stvml ***@***.***> wrote:

> Thanks for working on this, @RADXIshan <https://github.com/RADXIshan>! I
> think this could help with the intermittent database issues. Please look
> over my review comments when you get a chance.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/intel/cve-bin-tool/pull/4960#issuecomment-2762104119>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BNLSALJZT5LXBCDH5OMYBG32WWGXTAVCNFSM6AAAAABZS6KZ6GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRSGEYDIMJRHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: stvml]*stvml* left a comment (intel/cve-bin-tool#4960)
> <https://github.com/intel/cve-bin-tool/pull/4960#issuecomment-2762104119>
>
> Thanks for working on this, @RADXIshan <https://github.com/RADXIshan>! I
> think this could help with the intermittent database issues. Please look
> over my review comments when you get a chance.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/intel/cve-bin-tool/pull/4960#issuecomment-2762104119>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BNLSALJZT5LXBCDH5OMYBG32WWGXTAVCNFSM6AAAAABZS6KZ6GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRSGEYDIMJRHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

stvml reviewed (changes requested):
I think this is almost ready to merge. Please remove the `.DS_Store` files from your PR, and I'll run CI once those are gone. Thanks!

RADXIshan commented:
> I think this is almost ready to merge. Please remove the `.DS_Store` files from your PR, and I'll run CI once those are gone. Thanks!

I have removed '.DS_Store' from my PR, let me know if any other changes are needed

stvml commented on cve_bin_tool/cvedb.py:
```diff
@@ -764,17 +766,26 @@ def update_vendors(self, cve_data):
+    while retries < MAX_RETRIES:
+        try:
+            self.connection = sqlite3.connect(self.dbpath, timeout=10)  # Extend timeout for better handling
+            return self.connection.cursor()
```
Sorry, one last change here... I think we should keep the existing check for `if cursor is None` before we return the result of this call. Not sure if this error ever happens in the real world, but it's better to preserve logic that we aren't sure about :)

stvml reviewed (changes requested):
Looks like syntax errors are causing the tests to fail. You might try running the tests locally before pushing (see [here](https://github.com/intel/cve-bin-tool/blob/main/test/README.md) for documentation).

RADXIshan commented:
> Looks like syntax errors are causing the tests to fail. You might try running the tests locally before pushing (see [here](https://github.com/intel/cve-bin-tool/blob/main/test/README.md) for documentation).

One tab changed the whole indentation of the code and i didnt even realise :), thankss for the check i think now it will work 👍

stvml commented:
Please have a look at [this section of CONTRIBUTING.md](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#running-linters) for information about the linters this project uses. Highly recommend using `pre-commit` to automatically lint your contributions at push time.

RADXIshan commented:
> Please have a look at [this section of CONTRIBUTING.md](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#running-linters) for information about the linters this project uses. Highly recommend using `pre-commit` to automatically lint your contributions at push time.

ok i will go through it , this all is kind of new to me thats why i got confused , thanks for the guidance!

RADXIshan commented:
> Please have a look at [this section of CONTRIBUTING.md](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#running-linters) for information about the linters this project uses. Highly recommend using `pre-commit` to automatically lint your contributions at push time.

I have tried implementing as much as i understood and i think the errors from my end are fixed now , i went through the pre commit command and fixed the errors, it is showing some errors in some files which i have not touched ,like its showing that those files have not gone through much tests, but i fixed everything for my own file , can please check once and thanks a lot for guiding me through everything i really love working with you, lemme know if i am going wrong somewhere and how i can fix that :)

RADXIshan commented:
<img width="1440" alt="Screenshot 2025-04-09 at 2 55 28 AM" src="https://github.com/user-attachments/assets/2d1fe998-d0e5-4455-befd-c47431fb3dd0" />
i am not able to fix this can you please help me with it

terriko commented on cve_bin_tool/checkers/irssi.py:
```diff
@@ -20,7 +20,7 @@ class IrssiChecker(Checker):
         # r"# When irssi expands the templates in \"format\", the final string would be:",
         # r"# When irssi sees this kind of text, it goes to find \"name\" from abstracts",
-        # r"Log file \{hilight $0\} is locked, probably by another running Irssi",
+        # r"Log file \{highlight $0\} is locked, probably by another running Irssi",
```
Please remove this change from this PR.  It's not relevant and I'm not even sure it would be correct, since "hilight" may be the way it's written in the actual signature of irssi.

terriko reviewed (changes requested):
There's a whole pile of spelling fixes in here that should be in a separate PR, please.

RADXIshan commented:
> There's a whole pile of spelling fixes in here that should be in a separate PR, please.

Thanks for the review i will make a new pr on this 👍🏻

RADXIshan commented:
> There's a whole pile of spelling fixes in here that should be in a separate PR, please.

I have made a new pr , but now it seems like the issue is a little different

terriko reviewed (changes requested):
Yeah, sorry, I'm not sure how your branch got so messed up, but now you've got both spelling changes *and* a bunch of weird empty files. You may have to go search for some git tutorials and follow them carefully for now to re-create a PR on a new, clean branch that's synced with origin/main.  this is un-mergeable as is.

terriko commented:
For practice, I'd suggest you take ONE file with a spelling fix and make a PR just for that. Should help you figure out the flow and get you a useful fix that we can merge.

RADXIshan commented:
> Yeah, sorry, I'm not sure how your branch got so messed up, but now you've got both spelling changes _and_ a bunch of weird empty files. You may have to go search for some git tutorials and follow them carefully for now to re-create a PR on a new, clean branch that's synced with origin/main. this is un-mergeable as is.

I apologise for the mistake it might be due to my fork being old and not synced.Thank you for your time and patience, i will definitely look into it.

RADXIshan commented:
> For practice, I'd suggest you take ONE file with a spelling fix and make a PR just for that. Should help you figure out the flow and get you a useful fix that we can merge.

ok ok thanks, ill make a new fork and work on it

terriko commented:
I'm going to close this as it's currently un-mergeable, but feel free to open a new PR when you figure out how to get git to behave so you've got a change set that is either only the spelling fixes or only the database fixes and not a combination of both.
